src/module/requestManager.ts

```diff
@@ -321,13 +321,25 @@ function handleStatusCode(response: Response, errorStack?: unknown) {
 
   switch (status) {
     case HttpResponseCode.BadRequest:
+      throw new Error(
```

Comment:
Sadly, this would still not improve the stack frames showing where the error came from. I believe this should be changed slightly. Instead of `throw new Error()` this line would become `console.error(....)` and then below it we add `throw errorStack`

Reply:
side note: i am looking into how to get more than 3-4 stack frames i believe this might be something deno limits but v8 should give like 10 lines

Reply:
Noted, that should be done in another PR, again because it would be irrelevant to this PR. Title 🙄